src/main/java/com/mparticle/kits/InstabotKit.java

```diff
@@ -0,0 +1,164 @@
+package com.mparticle.kits;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.util.Log;
+
+import com.mparticle.MPEvent;
+import com.mparticle.MParticle;
+import com.rokolabs.sdk.tools.ThreadUtils;
+
+import java.util.List;
+import java.util.Map;
+
+import io.instabot.sdk.Instabot;
+import io.instabot.sdk.Settings;
+import io.instabot.sdk.account.Account;
+import io.instabot.sdk.analytics.Analytics;
+import io.instabot.sdk.analytics.Event;
+import io.instabot.sdk.push.Push;
+import io.instabot.sdk.push.PushConstants;
+import io.instabot.sdk.push.PushNotificationHelper;
+
+public class InstabotKit extends KitIntegration implements KitIntegration.EventListener, KitIntegration.PushListener, KitIntegration.AttributeListener, InstabotProvider {
+
+    @Override
+    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) {
+        if(!settings.containsKey("apiKey") || TextUtils.isEmpty(settings.get("apiKey"))){
+            Log.e("InstabotKit", "Instabot initialization was failed. Please set \"apiKey\".");
+            return null;
+        }
+
+        Instabot.start(settings.get("apiKey"), context, new Instabot.CallbackStart() {
+            @Override
+            public void start() {
+                Instabot.getSettings().setInstabotActivity("io.instabot.INSTABOT_ACTIVITY");
```

Comment:
do we need to instruct developers to add this to their manifest? If so, perhaps you could add it to the AndroidManifest.xml of this kit, or ideally in your SDK AAR.